deploy/modules/elasticaches/main.tf

```diff
@@ -16,33 +16,56 @@ resource "aws_kms_key" "cache_key" {
   enable_key_rotation = true
 }
 
-resource "aws_elasticache_serverless_cache" "cache" {
+resource "aws_elasticache_replication_group" "cache" {
   for_each = local.caches
-  
-  engine = "valkey"
-  name = "${var.environment}-${var.app}-${each.key}-cache"
 
-  cache_usage_limits {
-    data_storage {
-      maximum = var.cache_limits.data_storage_GB
-      unit = "GB"
-    }
+  replication_group_id = "${var.environment}-${var.app}-${each.key}-cache"
+  description          = "${var.environment} ${var.app} ${each.key} cache cluster"
 
-    ecpu_per_second {
-      maximum = var.cache_limits.ecpu_per_second
-    }
-  }
+  engine               = "valkey"
+  engine_version       = "8.0"
+  node_type            = var.node_type
+  port                 = 6379
+  parameter_group_name = aws_elasticache_parameter_group.cluster_enabled.name
 
-  daily_snapshot_time  = "02:00"
-  description          = "${var.environment} ${var.app} ${each.key} cache serverless cluster"
-  kms_key_id           = aws_kms_key.cache_key.arn
-  major_engine_version = "7"
-  security_group_ids   = [aws_security_group.cache_security_group.id]
+  num_node_groups         = 1 # 1 shard
+  replicas_per_node_group = 2 # 2 read replicas per shard
```

Comment:
Are the read replicas necessary?

Reply:
I think some redundancy in case a node fails is desirable, and 3 nodes are more than 1 but less than 5 😄

Reply:
btw, the current serverless cluster already replicates data in 3 AZs for redundancy

Reply:
everyone says 3 nodes is the magic number